File: frontend/lib/domain/repositories/memory_repository.dart
import '../entities/memory_entity.dart';

/// Repository interface - part of domain layer
/// Defines what operations are needed, not how they're implemented
abstract class MemoryRepository {
  Future<List<MemoryEntity>> getMemories({int? limit, int? offset});
  Future<MemoryEntity> getMemoryById(String id);
  Future<MemoryEntity> createTextMemory(String content, {String? title});
  Future<MemoryEntity> uploadAudio(String filePath);
  Future<MemoryEntity> uploadImage(String filePath);
  Future<void> deleteMemory(String id);
  Future<List<SearchResultEntity>> searchMemories(String query, {int limit = 10});
}

class SearchResultEntity {
  final MemoryEntity memory;
  final double score;

  SearchResultEntity({
    required this.memory,
    required this.score,
  });
}

File: backend/src/main/java/com/pocketmind/domain/repository/MemoryRepository.java
package com.pocketmind.domain.repository;

import com.pocketmind.domain.model.Memory;

import java.util.List;
import java.util.Optional;

/**
 * Repository interface - part of domain layer
 * Defines what operations are needed, not how they're implemented
 */
public interface MemoryRepository {
    List<Memory> findAll();
    List<Memory> findAllOrderByCreatedAtDesc();
    Optional<Memory> findById(Long id);
    Memory save(Memory memory);
    void deleteById(Long id);
    boolean existsById(Long id);
}

